src/analysis/Environments/Environments.ts

```diff
@@ -224,6 +222,7 @@ export const Environments = (props: EnvironmentsProps): ReactNode => {
       console.error('Pause is not currently implemented for azure apps');
     });
     await wrappedPauseCompute();
+    await refreshWorkspaces();
     await loadData();
   });
 
```

Comment:
what about the loadData call just below at around line 234?  There, the expectation was that a refresh data call would occur, but this PR would not make that call since it was shifted out of loadData funciton.

Reply:
also, when I skimmed through the code, this call appears to likely be causing a double-call on initial page/component load.  Seems we would want the useEffect to get modified so that only subsequent changes to shouldFilterByCreator trigger the effect, but not on first value.

Reply:
That `loadData` is called when the page is rendered, and uses the `useWorkspaces` hook to pull workspace data. When `refreshWorkspaces` was included, it was making 2 duplicate server calls to fetch the workspaces on page render, and the data was getting mangled.

Now, it just refreshes the workspaces when specific actions on the page are taken like pause/delete, and it makes sense to refresh the state.

Let me know if I'm thinking about this incorrectly though -- am kinda new to this code.

Reply:
I think I see what you mean… yeah we want subsequent changes to `shouldFilterByCreator` to trigger the workspace refresh, which is not happening currently. Will play around a bit more…